packages/api-plugin-fulfillment/src/mutations/createFulfillmentType.js

```diff
@@ -0,0 +1,44 @@
+import Random from "@reactioncommerce/random";
+import ReactionError from "@reactioncommerce/reaction-error";
+import { FulfillmentTypeSchema } from "../simpleSchemas.js";
+
+/**
+ * @method createFulfillmentType
+ * @summary Creates a new fulfillment type
+ * @param {Object} context - an object containing the per-request state
+ * @param {Object} input - Input object
+ * @param {String} input.name - name of fulfillment type
+ * @param {String} input.shopId - Shop Id
+ * @param {ProviderSchema} input.provider - Provider details
+ * @param {String} input.fulfillmentType - fulfillment type
+ * @param {String} input.displayMessageType - displayMessage for fulfillment type
+ * @param {FulfillmentMethodSchema[]} input.methods - ff-method array
+ * @returns {Promise<Object>} An object with the updated type
+ */
+export default async function createFulfillmentType(context, input) {
+  const cleanedInput = FulfillmentTypeSchema.clean(input);
+  if (!cleanedInput.provider) cleanedInput.provider = {};
+  cleanedInput.provider.name = cleanedInput.name;
+
+  if (cleanedInput.method) {
+    cleanedInput.method._id = Random.id();
+    cleanedInput.method.fulfillmentTypes = [cleanedInput.fulfillmentType];
+  }
```

Comment:
This `if` block seems unnecessary because the input has been cleaned and `method` isn't allowed, which means it will never exist.

Reply:
@aldeed 
Thanks for pointing out the issue.

The implementation was expected to add ff-method also (if details provided) along with adding a new ff-type. But as you pointed out, it had a bug and would not have worked s expected. 

Now after thinking about this for sometime, I have made the following changes.

I removed the `method` property from the `FulfillmentTypeSchema` . Now this mutation will only serve the purpose of creating the fulfillment-type and will not add any ff-method under it. For adding ff-methods, we should be using the mutation `createFulfillmentMethod` (as it makes sense to add a new ff-method only by a new plugin implementation)

@zenweasel - please let me know if the changes are okay.

Reply:
The description of the changes makes sense to me. I have not reviewed the code.

Reply:
I would be pushing the code shortly along with all the other changes. I just wanted a common understanding on the way we deal with user provided input.

Reply:
Changes are pushed to branch